api/storage/app/lighthouse-schema.graphql

```diff
@@ -832,7 +832,7 @@ input PoolBelongsToMany {
 
 type PoolCandidate {
   id: ID!
-  pool: Pool!
+  pool: Pool
```

Comment:
this is a rather significant type change, are we sure we want to do this?
table-wise, pool id is not nullable for a pool candidate

Reply:
I did this because the pool field is no longer available to any user (most users need to use poolAdvertisement instead) and it felt weird to me that we have a required field most users can't access. I'm not opposed to changing this back though if it's unpopular.

Reply:
Reverted with [193f025](https://github.com/GCTC-NTGC/gc-digital-talent/pull/5927/commits/193f02579a9d8422dad0101fbda3a777cac03ee0).